cypress/e2e/02-add_patient.cy.js

```diff
@@ -1,13 +1,24 @@
-import { generatePatient, loginHooks } from "../support/e2e";
+import { generatePatient, loginHooks, testNumber } from "../support/e2e";
+import { cleanUpPreviousOrg, setupOrgAndFacility } from "../utils/setup-utils";
 
 const patient = generatePatient();
 
 describe("Adding a single patient", () => {
   loginHooks();
   before("store patient info", () => {
+    // TODO: clean up after no tests rely on this patient
     cy.task("setPatientName", patient.fullName);
     cy.task("setPatientDOB", patient.dobForPatientLink);
     cy.task("setPatientPhone", patient.phone);
+    cy.task("getSpecName")
+      .then((prevSpecName) => {
+        if (prevSpecName) {
```

Comment:
Instead of just checking the spec name here, could we make an extra variable along the lines of `shouldCleanUpPreviousOrg` and spell out the logic as to why we would / wouldn't want to trigger the cleanup method? Right now in order to understand this code, we need the extra context that we're only cleaning up the previous org if there's been a previous run and think this code might be more maintainable if we could make that context explicit.

Also, Down the line if we reuse the `prevSpecName` state variable for something else, wouldn't want to couple that change with the cleanup getting triggered here, which wouldn't be immediately obvious to me. Maybe we can future-proof against that case with a more specific check in the variable assignment (ie move line 18 up to the top of the callback and set `shouldCleanUpPreviousOrg = prevSpecName === currentSpecName`

Reply:
ooo great callout -- let me think about this implementation more! Thank you for pointing this out!

Reply:
Updated based on your feedback! thank you!